src/session/session_helper.c

```diff
@@ -134,3 +134,35 @@ __wt_session_dump(WT_SESSION_IMPL *session, WT_SESSION_IMPL *dump_session, bool
     __wt_scr_free(session, &buf);
     return (ret);
 }
+
+/*
+ * __wt_session_set_last_error --
+ *     Stores information about the last error to occur during this session.
+ */
+int
+__wt_session_set_last_error(
+  WT_SESSION *wt_session, int err, int sub_level_err, const char *err_msg_content)
+{
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session = (WT_SESSION_IMPL *)wt_session;
+
+    char *err_msg = session->err_info.err_msg;
+    size_t err_msg_size;
+
+    /* Ensure arguments are valid, and that session is not a mock */
+    WT_ASSERT(session, __wt_is_valid_sub_level_error(sub_level_err));
+    if (err_msg_content == NULL || wt_session->get_last_error == NULL)
+        return (0);
```

Comment:
I'm not fond of returning 0, in the case that err_msg_content is NULL, that means a potential bug can leak into the system, without much warning. We should at least return WT_ERROR with a message

Reply:
I wrapped it in a WT_ASSERT in [Pass WT_SESSION_IMPL into __wt_session_set_last_error instead of WT_S…](https://github.com/wiredtiger/wiredtiger/pull/11361/commits/6fe3e3c5ccc5ba957a2fb8dab7b7992243af183b)